pkg/rancher-components/src/components/Form/ArrayList/ArrayList.vue

```diff
@@ -0,0 +1,347 @@
+          {{ title }}
+          <i v-if="showProtip" v-tooltip="protip" class="icon icon-info" />
+        </h3>
+      </slot>
+    </div>
+
+    <template v-if="rows.length">
+      <div v-if="showHeader">
+        <slot name="column-headers">
+          <label class="value text-label mb-10">
+            {{ valueLabel }}
+          </label>
+        </slot>
+      </div>
+      <div
+        v-for="(row, idx) in rows"
+        :key="idx"
+        class="box"
+        data-testid="array-list-box"
+      >
+        <slot
+          name="columns"
+          :i="idx"
+          :rows="rows"
+          :row="row"
+          :mode="mode"
+          :isView="isView"
+        >
+          <div class="value">
+            <slot
+              name="value"
+              :row="row"
+              :mode="mode"
+              :isView="isView"
+            >
+              <TextAreaAutoGrow
+                v-if="valueMultiline"
+                ref="value"
+                v-model="row.value"
+                :placeholder="valuePlaceholder"
+                :mode="mode"
+                :disabled="disabled"
+                @paste="onPaste(idx, $event)"
+                @input="update(idx, $event)"
+              />
+              <LabeledInput
+                v-else-if="rules.length > 0"
+                ref="value"
+                v-model="row.value"
+                :placeholder="valuePlaceholder"
+                :disabled="isView || disabled"
+                :rules="rules"
+                :compact="false"
+                @paste="onPaste(idx, $event)"
+                @input="update(idx, $event)"
+              />
+              <input
+                v-else
+                ref="value"
+                v-model="row.value"
+                :placeholder="valuePlaceholder"
+                :disabled="isView || disabled"
+                @paste="onPaste(idx, $event)"
+                @input="update(idx, $event)"
+              />
+            </slot>
+          </div>
+        </slot>
+        <div v-if="showRemove" class="remove">
+          <slot
+            name="remove-button"
+            :remove="() => remove(row, idx)"
+            :i="idx"
+            :row="row"
+          >
+            <button
+              type="button"
+              :disabled="isView"
+              class="btn role-link"
+              :data-testid="`remove-item-${idx}`"
+              @click="remove(row, idx)"
+            >
+              {{ removeLabel }}
+            </button>
+          </slot>
+        </div>
+      </div>
+    </template>
+    <div v-else-if="mode === 'view'" class="text-muted">&mdash;</div>
+    <div v-else>
+      <slot name="empty" />
+    </div>
+    <div v-if="showAdd && !isView" class="footer">
+      <slot v-if="showAdd" name="add" :add="add">
+        <button
+          data-testid="array-list-button"
+          type="button"
+          class="btn role-tertiary add"
+          :disabled="loading"
+          @click="add"
```

Comment:
@aalves08 pointed out that emitters do not automatically trigger functions in some cases.
Expected: `@click="add()`.